### ISSUE TITLE
field names with $ symbols result in errors

### ISSUE THREAD

**Issue by micimize** [google/built_value.dart#832]
https://github.com/google/built_value.dart/blob/e98d67d6f0896f781faa5307c066fa9dc270bbfa/built_value_generator/lib/src/value_source_class.dart#L963

Given an attribute named `$field`, will output something like:
```dart
  @override
  _$HeroForEpisodeResultHeroOnHuman build() {
    _$HeroForEpisodeResultHeroOnHuman _$result;
    try {
      _$result = _$v ??
          new _$HeroForEpisodeResultHeroOnHuman._($fields: $fields.build());
    } catch (_) {
      String _$failedField;
      try {
        _$failedField = '$fields';
        $fields.build();
      } catch (e) {
        throw new BuiltValueNestedFieldError(
            'HeroForEpisodeResultHeroOnHuman', _$failedField, e.toString());
      }
      rethrow;
    }
    replace(_$result);
    return _$result;
  }
}
```
Yielding an `Undefined name 'fields'.` error

**Comment by micimize** [comment]
I think this might only be an issue with builders, as it went away after switching to non-nested builders to work around #831

**Comment by micimize** [comment]
This was probably closed by [`v7.0.9`](https://github.com/google/built_value.dart/releases/tag/v7.0.9) will reopen if I discover I'm mistaken
